src/plugins/assetUploader/irys.ts

```diff
@@ -0,0 +1,115 @@
+import { FundResponse, IAssetUploader, UploadResponse, NodeIrys, CreateAndUploadOptions } from "./type";
+import { AptosConfig } from "../../api/aptosConfig";
+import { Account } from "../../core";
+import { Network } from "../../utils";
+
+/**
+ * Irys asset uploader
+ *
+ * Keeping the class here for now for simplicity, when/if extending the SDK
+ * to more providers encourged to get this out to its own file
+ */
+export class IrysAssetUploader implements IAssetUploader {
+  readonly config: AptosConfig;
+
+  private irysApp: typeof NodeIrys | undefined;
+
+  private constructor(config: AptosConfig, irysApp: typeof NodeIrys) {
+    this.config = config;
+    this.irysApp = irysApp;
+  }
+
+  static async init(config: AptosConfig) {
+    try {
+      const irys = await import("@irys/sdk");
+      return new IrysAssetUploader(config, irys.default);
+    } catch (e: any) {
+      throw new Error("To use the Irys service, please install the `@irys/sdk` package");
+    }
+  }
+
+  resolveProvider(): { irysNode: string; providerUrl: Network } {
+    switch (this.config.network) {
+      case Network.DEVNET:
+        // Irys does not support Aptos' devnet
+        throw new Error("Irys does not support Aptos' devnet");
+      case Network.TESTNET:
+        return { irysNode: "devnet", providerUrl: Network.TESTNET };
+      case Network.MAINNET:
+        // Irys supports node1 and node2, there is no major difference between
+        // those and it is recommended to simply use node1
+        return { irysNode: "node1", providerUrl: Network.MAINNET };
+      default:
+        throw new Error("Unsupported network");
```

Comment:
Could also just say `Irys does not support ${network}` including devnet, but this owrks as well.